userspace/ksud/src/su.rs

```diff
@@ -267,6 +270,26 @@ pub fn root_shell() -> Result<()> {
 
             set_identity(uid, gid, &groups);
 
+            // why is the rust representation of a 10-line c code like this on rs ??
+            use libc::{syscall, SYS_ioctl, SYS_readlinkat, SYS_setxattr, TCGETS};
+            let mut t = unsafe { std::mem::zeroed::<libc::termios>() };
+            let ioctl_result = unsafe {
+                syscall(SYS_ioctl, 0, TCGETS, &mut t)
+            };
+            
+            if ioctl_result == 0 {
+                let mut pts = vec![0u8; 64];
+                let ps = unsafe { syscall(SYS_readlinkat, libc::AT_FDCWD, CString::new("/proc/self/fd/0").unwrap().as_ptr(), pts.as_mut_ptr() as *mut libc::c_char, pts.len() as libc::c_ulong) };
+                
+                if ps != -1 {
+                    pts[ps as usize] = 0;
+                    let pts_path = str::from_utf8(&pts[..ps as usize]).unwrap_or_default();
+                    let ctx = CString::new("u:object_r:devpts:s0").unwrap();
```

Comment:
We can add a SELinux rule so that we can set con to `u:object_r:ksu_file:s0`

https://github.com/5ec1cff/KernelSU/blob/e8cf1271eabc338487b784d144e03473cf0273dc/kernel/selinux/rules.c#L29

```c
ksu_allow(db, KERNEL_SU_FILE, "devpts", "filesystem", "associate")
```